src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -69,6 +71,21 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         return userChallengesSet;
     }
 
+    @Override
+    public List<ChallengeDTO> getChallengesByCreatorId(Long creatorId) throws NotFoundException {
+
+        List<Challenge> challenges = challengeRepository.findAllByCreatorId(creatorId);
+
+        if(challenges == null || challenges.size() == 0){
```

Comment:
Aici ne putem folosi de ce ne spunea Alex pe grup: https://www.baeldung.com/java-optional

Reply:
Stiu despre ce este vorba ca am folosit si eu orElseThrow pt task-ul trecut. Aici este vorba de verificarea dimensiunii unei liste. Tot ce pot face este sa scriu asa  
return Optional.ofNullable(challengeDTOList.size() == 0 ? null: challengeDTOList).orElseThrow(()-> new NotFoundException().setMessage("Challenge answer not exists.")); si iar scriu linii de cod prea lungi.

Reply:
Tot in articolul ala sunt explicate 2 metode: filter() si map() pe care cred ca le poti folosi.